File: daily/2026/3010_Divide_an_Array_Into_Subarrays_With_Minimum_Cost_I.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int minimumCost(vector<int>& nums) {
        int s1 = INT_MAX;
        int s2 = INT_MAX;
        int ans = nums[0];
        for (int i = 1; i < nums.size(); i++){
            if (nums[i] < s2){
                s2 = max(nums[i], s1);
                s1 = min(nums[i], s1);
            }
        }
        ans += s1 + s2;
        return ans;
    }
};

int main(){
    vector<int> nums = {1,2,3,12};
    // vector<int> nums = {5,4,3};
    // vector<int> nums = {10,3,1,1};
    Solution S;
    cout << S.minimumCost(nums) << endl;
    return 0;
}

File: daily_python/3010_Divide_an_Array_Into_Subarrays_With_Minimum_Cost_I.py
import os
import sys
from typing import List

class Solution:
    def minimumCost(self, nums: List[int]) -> int:
        ans = nums[0]
        s1 = 100
        s2 = 100
        for i in nums[1:]:
            if i < s2:
                s2 = max(i, s1)
                s1 = min(i, s1)
        ans += s1 + s2
        return ans

def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            nums = list(map(int, lines[0].split('=')[1].strip()[1:-2].split(',')))
            dataset.append(nums)
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for nums in dataset:
        results.append(solution.minimumCost(nums))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')
    